src/Core/CustomRedirects/CustomRedirectCollection.cs

```diff
@@ -71,6 +80,14 @@ private CustomRedirect FindInternal(string url)
                 return redirect;
             }
 
+            // working with local copy to avoid multi-threading issues
```

Comment:
I think this is a wrong statement. `redirectsZA` is not a copy as an array is a reference type. This is a reference to the same data as `_redirectsZACache`.

Reply:
What I mean here is we never change that array, so can safely use it; but reference to that array can be changed during assignment from another thread, so we ensure we work with local array copy = local reference copy.